packages/doenetml-worker-rust/lib-doenetml-core/src/core/component_builder.rs

```diff
@@ -633,10 +637,15 @@ impl ComponentBuilder {
         let referent_children_virtual_node = self
             .structure_graph
             .get_component_children_virtual_node(referent_idx);
-        self.structure_graph.prepend_edge(
-            component_children_virtual_node,
-            referent_children_virtual_node,
-        );
+        // We create an intermediate virtual node so that we can mark it. This will indicate its descendants
+        // came from extending another component and aren't "original".
+        let intermediate_virtual_node = self.new_virtual_node();
```

Comment:
Just making sure I understand the new document structure. If a component1 is original, we still have

component1 -> component1-children-virtual-node -> component1's children

But, if we component2 is extending component1, we have
component2 -> component2-children-virtual-node -> marked-intermediate -> component-children-virtual-node
as well as
component2 -> component2-children-virtual-node -> component2's children

Reply:
Yep. You got it!